ui/src/App.tsx

```diff
@@ -15,11 +17,40 @@ const useStyles = makeStyles(() => ({
     },
 }));
 
+const api = new Api();
+
 const App = () => {
 
     const classes = useStyles();
 
     const [analyzeStatus, setAnalyzeStatus] = useState(null);
+    const [showTLSWarning, setShowTLSWarning] = useState(false);
+    const [userDismissedTLSWarning, setUserDismissedTLSWarning] = useState(false);
+    const [addressesWithTLS, setAddressesWithTLS] = useState(new Set());
+
+    useEffect(() => {
+        (async () => {
+            const recentTLSLinks = await api.getRecentTLSLinks();
+
+            if (recentTLSLinks?.length > 0) {
+                for (const tlsDestAddress of recentTLSLinks) {
+                    addressesWithTLS.add(tlsDestAddress)
+                }
+                setAddressesWithTLS(new Set(addressesWithTLS));
+                setShowTLSWarning(true);
+            }
+
+        })();
+    }, []);
+
+    const onTLSDetected = (destAddress: string) => {
+        addressesWithTLS.add(destAddress);
+        setAddressesWithTLS(new Set(addressesWithTLS));
```

Comment:
Why do we need two ways to `setAddressesWithTLS`?

Reply:
This is for notifications from websocket, the previous one is the result of the http request done on page load